packages/dd-trace/test/appsec/stack_trace.spec.js

```diff
@@ -0,0 +1,248 @@
+const path = require('path')
+
+const {
+  filterOutFramesFromLibrary,
+  reportStackTrace
+} = require('../../src/appsec/stack_trace')
+
+describe('Stack trace reporter', () => {
+  describe('filterOutFramesFromLibrary', () => {
+    it('should filer out frames from library', () => {
+      const callSiteList =
+        [...Array(10).keys()].map(i => (
+          {
+            getFileName: () => path.join(__dirname, `file${i}`),
+            getLineNumber: () => i,
+            getColumnNumber: () => i,
+            getFunctionName: () => `function${i}`
+          }
+        )
+        ).concat(
+          [...Array(10).keys()].map(i => (
+            {
+              getFileName: () => `file${i}`,
+              getLineNumber: () => i,
+              getColumnNumber: () => i,
+              getFunctionName: () => `function${i}`
+            }
+          ))
+        )
+
+      const expectedFrames = [...Array(10).keys()].map(i => (
+        {
+          file: `file${i}`,
+          line: i,
+          column: i,
+          function: `function${i}`
+        }
+      ))
+
+      const filteredFrames = filterOutFramesFromLibrary(callSiteList).map(frame => (
+        {
+          file: frame.getFileName(),
+          line: frame.getLineNumber(),
+          column: frame.getColumnNumber(),
+          function: frame.getFunctionName()
+        }
+      ))
+
+      assert.deepEqual(filteredFrames, expectedFrames)
+    })
+  })
+
+  describe('reportStackTrace', () => {
+    describe('report stack traces', () => {
+      const callSiteList = [...Array(20).keys()].map(i => (
+        {
+          getFileName: () => `file${i}`,
+          getLineNumber: () => i,
+          getColumnNumber: () => i,
+          getFunctionName: () => `function${i}`,
+          getTypeName: () => `type${i}`
+        }
+      ))
+
+      it('should not fail if no root span is passed', () => {
+        const rootSpan = undefined
+        const stackId = 'test_stack_id'
+        const maxDepth = 32
+        try {
+          reportStackTrace(rootSpan, stackId, maxDepth, 2, () => callSiteList)
+        } catch (e) {
+          assert.fail()
+        }
+      })
+
+      it('should add stack trace to rootSpan when meta_struct is not present', () => {
+        const rootSpan = {}
+        const stackId = 'test_stack_id'
+        const maxDepth = 32
+        reportStackTrace(rootSpan, stackId, maxDepth, 2, () => callSiteList)
+
+        const expectedFrames = [...Array(20).keys()].map(i => (
+          {
+            id: i,
+            file: `file${i}`,
+            line: i,
+            column: i,
+            function: `function${i}`,
+            class_name: `type${i}`
+          }
+        ))
+
+        assert.strictEqual(rootSpan.meta_struct['_dd.stack'].exploit[0].id, stackId)
+        assert.strictEqual(rootSpan.meta_struct['_dd.stack'].exploit[0].language, 'nodejs')
+        assert.deepEqual(rootSpan.meta_struct['_dd.stack'].exploit[0].frames, expectedFrames)
+      })
```

Comment:
This is just my preference, don't need to change, but I like to organise each test like prepare -> execute -> check:

```suggestion
      it('should add stack trace to rootSpan when meta_struct is not present', () => {
        const rootSpan = {}
        const stackId = 'test_stack_id'
        const maxDepth = 32
        const expectedFrames = [...Array(20).keys()].map(i => (
          {
            id: i,
            file: `file${i}`,
            line: i,
            column: i,
            function: `function${i}`,
            class_name: `type${i}`
          }
        ))

        reportStackTrace(rootSpan, stackId, maxDepth, 2, () => callSiteList)

        assert.strictEqual(rootSpan.meta_struct['_dd.stack'].exploit[0].id, stackId)
        assert.strictEqual(rootSpan.meta_struct['_dd.stack'].exploit[0].language, 'nodejs')
        assert.deepEqual(rootSpan.meta_struct['_dd.stack'].exploit[0].frames, expectedFrames)
      })
```

At least, I think that you should add a space before `reportStackTrace(rootSpan, stackId, maxDepth, 2, () => callSiteList)`

This applies to next tests also.